language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -174,12 +196,16 @@ private UserSettings(Map<String, Object> params) {
       this.disableTelemetry = (Boolean) params.getOrDefault(DISABLE_TELEMETRY, false);
     }
 
-    private static Map getAnalyzerProperties(Map<String, Object> params) {
-      Map analyzerProperties = (Map) params.get(ANALYZER_PROPERTIES);
-      if (analyzerProperties == null) {
-        analyzerProperties = Collections.emptyMap();
+    private Map<String, String> getAnalyzerProperties(Map<String, Object> params) {
+      try {
+        Map<String, String> analyzerProperties = (Map<String, String>) params.get(ANALYZER_PROPERTIES);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename "analyzerProperties" which hides the field declared at line 186. ([squid:HiddenFieldCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=192&issues=AWMr4Qgbau6S9RpGg66x&open=AWMr4Qgbau6S9RpGg66x)

[ISSUE_KEY]: AWMr4Qgbau6S9RpGg66x